### PR TITLE
Add terraform check

### PR DESCRIPTION
**Pull request by MancunianSam**
This runs terraform checks on pull requests to make sure the formatting and configuration are right.

### CODE REVIEW

File: .github/workflows/terraform_check.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        with:
+          role-to-assume: arn:aws:iam::${{ secrets.MANAGEMENT_ACCOUNT }}:role/TDRGithubTerraformAssumeRoleIntg
+          aws-region: eu-west-2
+          role-session-name: TerraformRole
+      - uses: hashicorp/setup-terraform@v1
+        with:
+          terraform_version: 1.1.3
+      - name: Terraform Format
+        id: fmt
+        run: terraform fmt -check
```
Comment (TomJKing) [inline .github/workflows/terraform_check.yml:34]: Does this not need the `-recursive` option?

File: .github/workflows/terraform_check.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      - name: Terraform Init
+        id: init
+        run: terraform init
+      - name: Select integration workspace
+        if: inputs.use-workspace
+        run: terraform workspace select intg
+      - name: Terraform Validate
+        id: validate
+        run: terraform validate -no-color
```
Comment (TomJKing) [inline .github/workflows/terraform_check.yml:43]: Will the `color` option work now we're not using Jenkins UI?

**Review by TomJKing — COMMENTED** [review]
Looks good. Just a couple of minor comments

File: .github/workflows/terraform_check.yml (hunk 2 above)
Comment (MancunianSam) [reply .github/workflows/terraform_check.yml]: Good point, I've taken the -no-color argument out.

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**